source/common/common/assert.h

```diff
@@ -322,5 +348,28 @@ void resetEnvoyBugCountersForTest();
 // but this macro replaces a less clear crash using NOT_REACHED_GCOVR_EXCL_LINE.
 #define PANIC_DUE_TO_CORRUPT_ENUM PANIC("corrupted enum");
 
+// CONDITION_STR is needed to prevent macros in condition from being expected, which obfuscates
+// the logged failure, e.g., "EAGAIN" vs "11".
+#define _ENVOY_NOTIFICATION_IMPL(CONDITION, NAME, DETAILS)                                         \
+  do {                                                                                             \
+    if (!(CONDITION)) {                                                                            \
+      const std::string& details = (DETAILS);                                                      \
+      ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::envoy_bug), debug,    \
```

Comment:
Should this have a different logger id than `envoy_bug`, or is that appropriate for how you plan to use this?

Reply:
Done